### ISSUE TITLE
Parallel query for different types of partitions does not work -  ERROR: "cannot modify commandid in active snapshot during a parallel operation"

### ISSUE THREAD

**Issue by rb26xpeople** [citusdata/citus#5949]
## The essence of the problem:
A query running in parallel in several sessions does not work if it looks at different types of partitions in the same table.

```sql
--select version();
--PostgreSQL 14.2 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44), 64-bit
--Citus - citus102_14 (sudo yum install -y citus102_14)

----Available: partitioned table
create table public.test_partition
(
    id     integer,
    datetime timestamp
)
    partition by RANGE (datetime);


--Partitioning by month
--y_2022_m_04 -> Columnar
create table public.test_partition_y_2022_m_04
    partition of public.test_partition
        FOR VALUES FROM ('2022-04-01 00:00:00') TO ('2022-05-01 00:00:00')
using columnar;

--y_2022_m_04 -> Heap
create table public.test_partition_y_2022_m_05
    partition of public.test_partition
        FOR VALUES FROM ('2022-05-01 00:00:00') TO ('2022-06-01 00:00:00');

--We insert a lot of data so that the table can run in parallel in the future
INSERT INTO public.test_partition
select
    generate_series(1, 10000000) AS id,
    to_date('010422', 'ddmmyy') + interval '1 day' * floor(random() * (60 + 1));

--It works. Since it works out in 1 session
select * from public.test_partition where id = 1 ;

--If you have parallel query execution configured in your PostgreSQL configuration, this query will return an error:
--"cannot modify commandid in active snapshot during a parallel operation"
create table public.check as select * from public.test_partition where id = 1 ;

--If you force a request to be sent to one partition, it will work even in parallel.
create table public.check as select * from public.test_partition where datetime  between '2022-05-01' and '2022-05-31' ;
```

**Comment by onurctirtir** [comment]
Same as https://github.com/citusdata/citus/issues/5270

**Comment by onurctirtir** [comment]
Same as https://github.com/citusdata/citus/issues/5270

**Comment by emelsimsek** [comment]
Duplicate of https://github.com/citusdata/citus/issues/5270
